### PR TITLE
Add ability to ignore specific warnings via magic comments

### PR DESCRIPTION
**Pull request by japhib**
Initial implementation for https://github.com/esl/gradient/issues/160

### CODE REVIEW

**Comment by japhib** [comment]
@erszcz Finally got checks passing! Turns out it was failing because an ignore in `.gradient_ignore.exs` needed to be edited with a changed line number -- which, coincidentally, is exactly the issue this PR is built to resolve 😅

To help prevent issues like this with self-gradualizing in the future, I've gone through the warnings in `.gradient_ignore.exs`, removed a couple that appear to no longer be needed, and changed some other ones for specific lines into magic comments on those specific lines.

File: lib/gradient/ast_specifier.ex [hunk 1]
```diff
@@ -1,10 +1,10 @@
 defmodule Gradient.AstSpecifier do
   @moduledoc """
   Module adds missing location information to the Erlang abstract code produced
-  from Elixir AST. Moreover it can be used to catch some ast pattern and replace 
+  from Elixir AST. Moreover it can be used to catch some ast pattern and replace
```
Comment (japhib) [inline lib/gradient/ast_specifier.ex:4]: My editor really dislikes the trailing whitespace on these lines. If desired, I can undo these changes to clean up the PR a bit.

File: .gradient_ignore.exs [hunk 2]
```diff
@@ -1,19 +1,6 @@
...
-
-  "lib/gradient/elixir_type.ex:38",
-
-  "lib/gradient/ast_specifier.ex:439",
-
-  "lib/gradient.ex:77",
-
-  "lib/gradient.ex:184",
-
-  "lib/mix/tasks/gradient.ex:106",
```
Comment (japhib) [inline .gradient_ignore.exs:15]: Some of these were changed into magic comments in the files themselves. Others (such as the one on line 5) didn't result in an error when I commented out the ignore, so I just removed them.

File: lib/gradient.ex [hunk 3]
```diff
@@ -211,18 +222,4 @@ defmodule Gradient do
       forms -> {:ok, forms}
     end
   end
-
-  defp filter_ignore_paths(errors, opts) do
```
Comment (japhib) [inline lib/gradient.ex:215]: Moved to `Error.reject_errors_in_ignore_paths`

File: lib/gradient/ast_specifier.ex (hunk 1 above)
Comment (erszcz) [reply lib/gradient/ast_specifier.ex:4]: That's fine, I don't like these either, but they somehow sneaked in :D

**Comment by japhib** [comment]
No worries, thanks for merging!

**Pull request merged**